src/index.ts

```diff
@@ -130,92 +121,64 @@ export function resolveInput(
 async function resolveTarStreamOnEntry(
 	header: tar.Headers,
 	stream: Readable,
-	bundle: Bundle,
 	resolvers: Resolver[],
 	pack: tar.Pack,
-	specifiedFileResolver: null | Resolver = null,
 	dockerfile?: string,
-): Promise<Resolver | undefined> {
-	const name = TarUtils.normalizeTarEntry(header.name);
-
+): Promise<void> {
+	const name = header.name ? TarUtils.normalizeTarEntry(header.name) : '';
 	if (!name) {
 		await TarUtils.drainStream(stream);
 		return;
 	}
-
-	// Handle the case where the dockerfile is specified
-	if (dockerfile != null) {
-		if (specifiedFileResolver == null && name === dockerfile) {
-			specifiedFileResolver = await resolveSpecifiedFile(
-				resolvers,
-				bundle,
-				name,
-				header,
-				stream,
-				pack,
-			);
-			return specifiedFileResolver;
+	const potentials = resolvers.filter(r =>
```

Comment:
NIT: *candidates* is a slightly better identifier for this.

Reply:
Done. In addition to this change, I've also added code comments to each of the 4 `needsEntry` methods -- the same comments I had added to the PR.